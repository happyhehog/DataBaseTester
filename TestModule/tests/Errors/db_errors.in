[@PrintLevel] MAIN

@@ ================|
@@ Database errors |
@@ ================|

[@Clear]
[@NoOutput]
CREATE TABLE t(id INT);
[@NoOutput]
CREATE TABLE t(id INT);

[@Clear]
CREATE TABLE t(id INT, id1 FLOAT NOT NULL NOT NULL);

[@Clear]
CREATE TABLE t(id INT, id CHAR(20));

[@Clear]
[@NoOutput]
CREATE TABLE t(id INT UNIQUE);
INSERT INTO t VALUES(1);
INSERT INTO t VALUES(2);
INSERT INTO t VALUES(3);
[@NoOutput]
INSERT INTO t VALUES(1);
UPDATE t SET id = 1 WHERE id = 3;
SELECT * FROM t JOIN t1 on t.id = t1.id;

[@Clear]
[@NoOutput]
CREATE TABLE t(id INT, num FLOAT NOT NULL);
[@NoOutput]
INSERT INTO t(id) VALUES(1);
INSERT INTO t(id, id) VALUES(2, 2);
INSERT INTO tt VALUES(1, 2);
INSERT INTO t VALUES(1, 2, 3);
INSERT INTO t VALUES(99999999999999999999999, 1);
INSERT INTO t(id, temp) VALUES(1, 2);
INSERT INTO t VALUES(1, 1, 1);
INSERT INTO t VALUES('asdasd', 123);
UPDATE t SET num = null WHERE id = 1;

[@Clear]
[@NoOutput]
CREATE TABLE t(id INT, s CHAR(5));
[@NoOutput]
INSERT INTO t VALUES(1, 'qweasdzxc');
