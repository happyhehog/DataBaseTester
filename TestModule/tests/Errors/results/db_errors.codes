{"status":"DB_ERROR","command":"CREATE TABLE t(id INT);","answer":"Table already exists ERROR: 1"}
{"status":"DB_ERROR","command":"CREATE TABLE t(id INT, id1 FLOAT NOT NULL NOT NULL);","answer":"Constraint already exists ERROR: 3"}
{"status":"DB_ERROR","command":"CREATE TABLE t(id INT, id CHAR(20));","answer":"Field name already used ERROR: 4"}
{"status":"DB_ERROR","command":"INSERT INTO t VALUES(1);","answer":"Value already exists, use UNIQUE value ERROR: 9"}
{"status":"DB_ERROR","command":"UPDATE t SET id = 1 WHERE id = 3;","answer":"Value already exists, use UNIQUE value ERROR: 9"}
{"status":"DB_ERROR","command":"SELECT * FROM t JOIN t1 on t.id = t1.id;","answer":"Table doesn`t exist ERROR: 2"}
{"status":"DB_ERROR","command":"INSERT INTO t(id) VALUES(1);","answer":"Null values unavailable ERROR: 8"}
{"status":"DB_ERROR","command":"INSERT INTO t(id, id) VALUES(2, 2);","answer":"Same column in request ERROR: 7"}
{"status":"DB_ERROR","command":"INSERT INTO tt VALUES(1, 2);","answer":"Table doesn`t exist ERROR: 2"}
{"status":"DB_ERROR","command":"INSERT INTO t VALUES(1, 2, 3);","answer":"Invalid count of columns and values ERROR: 5"}
{"status":"OK","command":"INSERT INTO t VALUES(99999999999999999999999, 1);","answer":"Int oversize (Str num 1, sym num 41): 99999999999999999999999"}
{"status":"DB_ERROR","command":"INSERT INTO t(id, temp) VALUES(1, 2);","answer":"Field doesnt exist ERROR: 6"}
{"status":"DB_ERROR","command":"INSERT INTO t VALUES(1, 1, 1);","answer":"Invalid count of columns and values ERROR: 5"}
{"status":"DB_ERROR","command":"INSERT INTO t VALUES('asdasd', 123);","answer":"Value is out of range OR is not a number ERROR: 12"}
{"status":"DB_ERROR","command":"UPDATE t SET num = null WHERE id = 1;","answer":"Null values unavailable ERROR: 8"}
{"status":"DB_ERROR","command":"INSERT INTO t VALUES(1, 'qweasdzxc');","answer":"CHAR IS VERY BIG ERROR: 13"}
