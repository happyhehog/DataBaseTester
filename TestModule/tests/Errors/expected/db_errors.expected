Table already exists ERROR: 1
Constraint already exists ERROR: 3
Field name already used ERROR: 4
Value already exists, use UNIQUE value ERROR: 9
Value already exists, use UNIQUE value ERROR: 9
Table doesn`t exist ERROR: 2
Null values unavailable ERROR: 8
Same column in request ERROR: 7
Table doesn`t exist ERROR: 2
Invalid count of columns and values ERROR: 5
Int oversize (Str num 1, sym num 41): 99999999999999999999999
Field doesnt exist ERROR: 6
Invalid count of columns and values ERROR: 5
Value is out of range OR is not a number ERROR: 12
Null values unavailable ERROR: 8
CHAR IS VERY BIG ERROR: 13
